### ISSUE TITLE
Improvement suggestions to pin_ctrl() & oin_ctrl()

### ISSUE THREAD

**Issue by janzzon** [rOpenGov/sweidnumbr#90]
The functions pin_ctrl() & oin_ctrl() seems to be designed to require that input values are correct formatted. 
```pin_ctrl()``` can either throw NA or FALSE if the input is wrong. Suggestion is that anything that isn't a valid pin should return FALSE?
``` r
# Sample values to check
num_to_check <- c("202100-6255","121212-1212","19121212-1212","121212+1212","1212121212",
                  1212121212, NA, Inf, TRUE, F, "foo", 123, 456L)
pin_ctrl(num_to_check)
#> Assumption: Pin of format YYMMDDNNNC is assumed to be less than 100 years old
#> Warning in as.pin.character(pin): Erroneous pin(s) (set to NA).
#>  [1]   NA TRUE TRUE TRUE TRUE TRUE   NA   NA   NA   NA   NA   NA   NA
```

Function ```oin_ctrl()``` stops with an error when invalid input is provided. Result of the failed check is only provided as a warning message, which is hard to make anything useful with. No normal output is provided from the function.
``` r
oin_ctrl(num_to_check)
#> Warning: The following personal identity numbers are incorrect: 2, 3, 4, 5,
#> 6, 7, 8, 9, 10, 11, 12, 13
#> Error in FUN(X[[i]], ...): All elements of x must have the same length with 9-12 digits!
```
Suggestion is the same as for ```pin_ctrl()```, valid oin should return TRUE, everything else FALSE.

#### Example wrapper functions
I wrote 2 simple wrapper functions to show my suggestions:
``` r
# Sample values to check
num_to_check <- c("202100-6255","121212-1212","19121212-1212","121212+1212","1212121212",
                  1212121212, NA, Inf, TRUE, F, "foo", 123, 456L)

# Wrapper function to set failing pin input to FALSE and valid to TRUE
pin_ctrl2 <- Vectorize({
  function(x) {
    ifelse (!is.na(try(suppressWarnings(suppressMessages(pin_ctrl(x))),silent = TRUE)),
            pin_ctrl(x), FALSE)
  }
})

# Wrapper function to set failing oin input to FALSE and valid to TRUE
oin_ctrl2 <- Vectorize({
  function(x) {
    ifelse ((class(try(suppressWarnings(oin_ctrl(x)), silent =TRUE)) != "try-error"),
            oin_ctrl(x), FALSE)
  }
})

# Are values pin?
pin_ctrl2(num_to_check)
#> Assumption: Pin of format YYMMDDNNNC is assumed to be less than 100 years old
#> Assumption: Pin of format YYMMDDNNNC is assumed to be less than 100 years old
#> Assumption: Pin of format YYMMDDNNNC is assumed to be less than 100 years old
#> Assumption: Pin of format YYMMDDNNNC is assumed to be less than 100 years old
#>   202100-6255   121212-1212 19121212-1212   121212+1212    1212121212 
#>         FALSE          TRUE          TRUE          TRUE          TRUE 
#>    1212121212          <NA>           Inf          TRUE         FALSE 
#>          TRUE         FALSE         FALSE         FALSE         FALSE 
#>           foo           123           456 
#>         FALSE         FALSE         FALSE

# Are values oin?
oin_ctrl2(num_to_check)
#>   202100-6255   121212-1212 19121212-1212   121212+1212    1212121212 
#>          TRUE         FALSE         FALSE         FALSE         FALSE 
#>    1212121212          <NA>           Inf          TRUE         FALSE 
#>         FALSE         FALSE         FALSE         FALSE         FALSE 
#>           foo           123           456 
#>         FALSE         FALSE         FALSE
```

If the functionality above doesn't breaks any existing code, may it be possible to add this behaviour to the functions as default?

**Comment by MansMeg** [comment]
It is a good idea, but I think it would be better to include an extra argument in ```pin_ctrl()``` called ```force_logical=FALSE``` and if this argument is flagged to ```TRUE``` it will only return a logical vector without any ```NA```:s, as you suggest. Then the function contain this functionality but the package is still back compatible.

**Comment by janzzon** [comment]
Extra argument to both functions sounds good!
The functionality i lack is to easily be able to verify which values in a
table that are pin, oin or simply bad data.



lör 23 jan 2016 14:55 Måns Magnusson <notifications@github.com> skrev:

**Comment by MansMeg** [comment]
Alright. Then adding those two arguments should help you out. I will not have time to fix this right away, so you'll have to live with the current functionality until next update.

**Comment by MansMeg** [comment]
This is now fixed with version 1.1.0 @ github. Install with devtools::install_github("rOpenGov/sweidnumbr").
